packages-common/pluggable-widgets-tools/tests/commands.js

```diff
@@ -0,0 +1,175 @@
+            mkdir(workDir);
+
+            if (version === "latest") {
+                let generatedWidget;
+                const release = await mutex.acquire(); // yeoman generator is no re-entrable :(
+                try {
+                    generatedWidget = await runYeoman(require.resolve("@mendix/generator-widget/generators/app"))
+                        .inTmpDir()
+                        .withPrompts({
+                            widgetName: "Generated",
+                            programmingLanguage: lang === "ts" ? "typescript" : "javascript",
+                            platform,
+                            e2eTests: false
+                        })
+                        .withArguments("Generated")
+                        .toPromise();
+                } finally {
+                    release();
+                }
+                await copy(join(generatedWidget, "generated"), workDir);
+            } else {
+                await copy(join(__dirname, "projects", widgetName), workDir);
+            }
+
+            const widgetPackageJson = await readJson(join(workDir, "package.json"));
+            widgetPackageJson.devDependencies["@mendix/pluggable-widgets-tools"] = toolsPackagePath;
+            await writeJson(join(workDir, "package.json"), widgetPackageJson);
+
+            await execAsync("npm install --loglevel=error", workDir);
+        }
+
+        async function testLint() {
+            await execFailedAsync("npm run lint", workDir);
+            await execAsync("npm run lint:fix", workDir);
+            await execAsync("npm run lint", workDir);
+        }
+
+        async function testTest() {
+            await execFailedAsync("npm test", workDir);
+            await execAsync("npm test -- -u", workDir);
+        }
+
+        async function testStart() {
+            const startProcess = exec("npm start", { cwd: workDir });
+
+            return Promise.race([
+                new Promise((_, reject) => setTimeout(() => reject(new Error("Timeout!")), 100000)),
+                new Promise((resolve, reject) => {
+                    startProcess.stdout.on("data", data => {
+                        if (/\berror\b/i.test(data)) {
+                            reject(new Error(`Received error ${data}`));
+                        } else if (data.includes("Finished 'copyToDeployment'")) {
+                            console.log(`[${widgetName}] Done!`);
+                            resolve();
+                        }
+                    });
+                    startProcess.stderr.on("data", data => {
+                        reject(new Error(`Received error output: ${data}`));
+                    });
+                    startProcess.on("exit", code => {
+                        reject(new Error(`Exited with status ${code}`));
+                    });
+                })
+            ]).finally(() => fkill(startProcess.pid, { force: true, tree: true, silent: true }));
```

Comment:
You can rewrite this to a try - finally clause

Reply:
👍